core/dbt/include/global_project/macros/materializations/models/table/columns_spec_ddl.sql

```diff
@@ -1,19 +1,21 @@
-{%- macro get_columns_spec_ddl() -%}
-  {{ adapter.dispatch('get_columns_spec_ddl', 'dbt')() }}
+{%- macro get_table_columns_and_constraints() -%}
+  {{ adapter.dispatch('get_table_columns_and_constraints', 'dbt')() }}
 {%- endmacro -%}
 
-{% macro default__get_columns_spec_ddl() -%}
-  {{ return(columns_spec_ddl()) }}
+{% macro default__get_table_columns_and_constraints() -%}
+  {{ return(table_columns_and_constraints()) }}
 {%- endmacro %}
 
-{% macro columns_spec_ddl() %}
+{% macro table_columns_and_constraints() %}
   {# loop through user_provided_columns to create DDL with data types and constraints #}
-    {%- set user_provided_columns = model['columns'] -%}
+    {%- set raw_column_constraints = adapter.render_raw_columns_constraints(raw_columns=model['columns']) -%}
+    {%- set raw_model_constraints = adapter.render_raw_model_constraints(raw_constraints=model['constraints']) -%}
     (
-    {% for i in user_provided_columns %}
-      {%- set col = user_provided_columns[i] -%}
-      {%- set constraints = col['constraints'] -%}
-      {{ col['name'] }} {{ col['data_type'] }}{% for c in constraints %} {{ adapter.render_raw_column_constraint(c) }}{% endfor %}{{ "," if not loop.last }}
+    {% for c in raw_column_constraints -%}
+      {{ c }}{{ "," if not loop.last or raw_model_constraints }}
```

Comment:
Can we trust Jinja to reliably treat `Null` as `false`? Can there be weird edge cases like whitespace messing up logic things?

Reply:
We don't have a standard, and give him some ginger I actually brought this morning, I do think that `None` (whatever name they've changed it to for this package) is falsey. 

For context, I was especially concerned about an empty str and being unclear on how it's truthiness would be evaluated, but we've eliminated that as a possibility with your recent changes so I think this is less of a concern than it was for me before